GVFS/FastFetch/FastFetchLibGit2Repo.cs

```diff
@@ -0,0 +1,72 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Prefetch.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace FastFetch
+{
+    public class FastFetchLibGit2Repo : LibGit2Repo
+    {
+        public FastFetchLibGit2Repo(ITracer tracer, string repoPath)
+            : base(tracer, repoPath)
+        {
+        }
+
+        public virtual bool TryCopyBlobToFile(string sha, IEnumerable<PathWithMode> destinations, out long bytesWritten)
+        {
+            IntPtr objHandle;
+            if (Native.RevParseSingle(out objHandle, this.RepoHandle, sha) != Native.SuccessCode)
+            {
+                bytesWritten = 0;
+                EventMetadata metadata = new EventMetadata();
+                metadata.Add("ObjectSha", sha);
+                this.Tracer.RelatedError(metadata, "Couldn't find object");
+                return false;
+            }
+
+            try
+            {
+                // Avoid marshalling raw content by using byte* and native writes
+                unsafe
+                {
+                    switch (Native.Object.GetType(objHandle))
+                    {
+                        case Native.ObjectTypes.Blob:
+                            byte* originalData = Native.Blob.GetRawContent(objHandle);
+                            long originalSize = Native.Blob.GetRawSize(objHandle);
+
+                            foreach (PathWithMode destination in destinations)
+                            {
+                                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))

```

Comment:
`IsOSPlatform` performs a string comparison:

*IsOSPlatform(Mac)*
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Unix.cs#L22

*IsOSPlatform(Windows)*
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Windows.cs#L19

*OSPlatform.Equals*
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs#L37

And so we should switch to setting either a `bool` or an `Action<...>` in the constructor to avoid calling `IsOSPlatform` for every blob.

Reply:
Great catch, thanks! I moved this into the constructor.